core/src/main/java/org/elasticsearch/transport/TransportService.java

```diff
@@ -447,6 +448,27 @@ public void removeConnectionListener(TransportConnectionListener listener) {
         sendRequest(node, action, request, TransportRequestOptions.EMPTY, handler);
     }
 
+    public <T extends TransportResponse> void sendChildRequest(final DiscoveryNode node, final String action,
+                                                               final TransportRequest request, final Task parentTask,
+                                                               final TransportResponseHandler<T> handler) {
+        sendChildRequest(node, action, request, parentTask, TransportRequestOptions.EMPTY, handler);
+    }
+
+    public <T extends TransportResponse> void sendChildRequest(final DiscoveryNode node, final String action,
+                                                               final TransportRequest request, final Task parentTask,
+                                                               final TransportRequestOptions options,
+                                                               final TransportResponseHandler<T> handler) {
+        request.setParentTask(localNode.getId(), parentTask.getId());
+        try {
+            taskManager.registerChildTask(parentTask, node.getId());
+        } catch (IllegalStateException ex) {
+            // The parent task is already cancelled - just fail the request
+            handler.handleException(new TransportException(ex));
+            return;
+        }
+        sendRequest(node, action, request, options, handler);
```

Comment:
nit: I'd prefer that we move the sendRequest call to the try block and remove the return from the catch block